WNPRC_Purchasing/src/client/model.ts

```diff
@@ -0,0 +1,110 @@
+import { immerable } from 'immer';
+
+export class RequestOrderModel {
+    /**
+     * @hidden
+     */
+    [immerable] = true;
+
+    readonly account: string;
+    readonly accountOther: string;
+    readonly vendor: string; //rowId of ehr_purchasing.vendor
+    readonly newVendor?: VendorModel = VendorModel.create({});
+    readonly purpose: string;
+    readonly shippingDestination: number; //rowId of ehr_purchasing.shippingInfo
+    readonly deliveryAttentionTo: string;
+    readonly comments?: string;
+    readonly errorMsg?: string;
+    readonly errors?: any;
+    readonly qcState?: number;
+
+    constructor(values?: Partial<RequestOrderModel>) {
+        Object.assign(this, values);
+    }
+
+    static create(raw: any, defaultSettings = null): RequestOrderModel {
```

Comment:
Can you make the raw parameter optional on all these models or at least have a default value?  There are a lot of models created like RequestOrderModel.create({}) instead of RequestOrderModel.create().  The empty object parameter isn't really necessary.

Reply:
done